### PR TITLE
policy validation: don't try so hard

### LINKED ISSUE #94: panic when `policy` does not include `strategy`
for empty scaling policy in nomad job: 
```
goroutine 85 [running]:
github.com/hashicorp/nomad-autoscaler/policystorage.validate(0xc0003c21e0, 0xc00071c060, 0x24, 0x0)
	/Users/cgbaker/go/src/github.com/hashicorp/nomad-autoscaler/policystorage/nomad.go:103 +0x334
github.com/hashicorp/nomad-autoscaler/policystorage.(*Nomad).Get(0xc00000e028, 0xc00071c060, 0x24, 0x0, 0x0, 0x0)
	/Users/cgbaker/go/src/github.com/hashicorp/nomad-autoscaler/policystorage/nomad.go:38 +0xa8
github.com/hashicorp/nomad-autoscaler/agent.(*Agent).Run.func1(0xc00078b730, 0x195b840, 0xc00020c500, 0xc00000e028, 0x1965c60, 0xc000717f80, 0xc00010e480, 0xc00071c060, 0x24)
	/Users/cgbaker/go/src/github.com/hashicorp/nomad-autoscaler/agent/agent.go:84 +0xa6
created by github.com/hashicorp/nomad-autoscaler/agent.(*Agent).Run
	/Users/cgbaker/go/src/github.com/hashicorp/nomad-autoscaler/agent/agent.go:78 +0x60f
```

### PR DESCRIPTION
**Pull request by cgbaker**
resolves #94 

policy was nobly trying to validate additional errors after discovery that "strategy" was of the wrong type (or, in the case of #94, did not exist). early return from this case.

### CODE REVIEW

File: policystorage/nomad.go [hunk 1]
```diff
@@ -98,6 +98,7 @@ func validate(policy *api.ScalingPolicy) []error {
 	strategyList, ok := policy.Policy["strategy"].([]interface{})
```
Comment (cgbaker) [inline policystorage/nomad.go:98]: question: why is this `[]interface{}` (i.e., []map[string]interface{} below)? why does this work?

**Comment by cgbaker** [comment]
> As a future item, I am curious if we can just remove the `[]error` from the validate function. I don't believe it will ever return > 1 error with this change.

I was thinking the same thing. I imagine it's probably cribbed from Nomad, where plan/validate can tell you everything that's wrong with your job. But the function as written doesn't have that depth of capability.

File: policystorage/nomad.go (hunk 1 above)
Comment (cgbaker) [reply policystorage/nomad.go:98]: @jrasell , @lgfa29 : what about this? ^

Comment (cgbaker) [reply policystorage/nomad.go:98]: i'm guessing it's some weird Go thing I didn't know about?

Comment (jrasell) [reply policystorage/nomad.go:98]: I am honestly not sure. Unless we plan/planned on having scaling policies support a list of strategies, I believe we could just perform the `map[string]interface{}` cast. This is all conjecture without testing though.

Comment (lgfa29) [reply policystorage/nomad.go:98]: `hcl.DecodeObject` [here](https://github.com/hashicorp/nomad/blob/master/jobspec/parse_group.go#L350) turns `strategy` into a list.

**Comment by lgfa29** [comment]
> > As a future item, I am curious if we can just remove the `[]error` from the validate function. I don't believe it will ever return > 1 error with this change.
> 
> I was thinking the same thing. I imagine it's probably cribbed from Nomad, where plan/validate can tell you everything that's wrong with your job. But the function as written doesn't have that depth of capability.

Validations often return multiple errors, that's why I used a list here. We currently don't have a case where multiple errors would be returned, but it could happen in the future. Using [go-multierror](https://github.com/hashicorp/go-multierror) would probably be better.

**Pull request merged**